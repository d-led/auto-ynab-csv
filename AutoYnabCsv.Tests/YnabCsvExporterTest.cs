using ApprovalTests;
using ApprovalTests.Reporters;
using AutoYnabCsv.Contracts;
using AutoYnabCsv.Exporters;

namespace Tests;

[TestClass]
[UseReporter(typeof(DiffReporter))]
public class YnabCsvExporterTest
{
    [TestMethod]
    public void ExportingEmptyCsv()
    {
        Approvals.Verify(YnabCsvExporter.Export([]));
    }

    [TestMethod]
    public void ExportingNonEmptyCsv()
    {
        Approvals.Verify(YnabCsvExporter.Export([
            new YnabImportEntry(
                Date: new DateOnly(2024, 12, 22), 
                Payee: "Some payee",
                 Memo: "memo1",
                 Inflow: 0,
                Outflow: 2.5m),
            new YnabImportEntry(
                Date: new DateOnly(2024, 1, 23), 
                Payee: "Payee2",
                Memo: "memo with comma, here",
                Inflow: 3500.0m,
                Outflow: 0),
        ]));
    }
}