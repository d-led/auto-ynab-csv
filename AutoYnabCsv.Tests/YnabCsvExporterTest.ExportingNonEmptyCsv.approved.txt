Date,Payee,Memo,Inflow,Outflow
2024-12-22,Some payee,memo1,0,2.5
2024-01-23,Payee2,"memo with comma, here",3500.0,0
